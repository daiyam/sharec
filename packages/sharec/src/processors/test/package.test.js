const { vol } = require('memfs')
const {
  processPackageJson,
  clearPackageJson,
  getCurrentPackageJsonMetaData,
} = require('../package')

describe('processors > package >', () => {
  const packageJson01 = require('fixtures/package/package_01.json')
  const packageJson02 = require('fixtures/package/package_02.json')
  const packageJson5 = require('../../../test/fixtures/package/package_05.json')
  const packageJson6 = require('../../../test/fixtures/package/package_06.json')

  beforeEach(() => {
    vol.reset()
  })

  describe('processPackageJson', () => {
    it('should merge exist package.json with upcoming from configs', async () => {
      expect.assertions(1)

      const dir = {
        '/target/package.json': JSON.stringify(packageJson01, null, 2),
        '/configs/package.json': JSON.stringify(packageJson02, null, 2),
      }
      vol.fromJSON(dir, '/')

      await processPackageJson('/configs', '/target')

      expect(
        JSON.parse(vol.readFileSync('/target/package.json', 'utf8')),
      ).toMatchSnapshot()
    })
  })

  describe('getCurrentPackageJsonMetaData', () => {
    it('should extract sharec meta data from target project package.json', async () => {
      expect.assertions(1)

      const metaData = {
        injected: true,
      }
      const dir = {
        '/target/package.json': JSON.stringify(
          {
            sharec: metaData,
          },
          null,
          2,
        ),
      }
      vol.fromJSON(dir, '/')

      const receivedMetaData = await getCurrentPackageJsonMetaData('/target')

      expect(receivedMetaData).toEqual(metaData)
    })

    it('should return null if sharec meta data is not exist', async () => {
      expect.assertions(1)

      const dir = {
        '/target/package.json': JSON.stringify({}, null, 2),
      }
      vol.fromJSON(dir, '/')

      const receivedMetaData = await getCurrentPackageJsonMetaData('/target')

      expect(receivedMetaData).toEqual(null)
    })
  })

  describe('clearPackageJson', () => {
    it('should remove configs from package json and injection status', async () => {
      const dir = {
        '/target/package.json': JSON.stringify(packageJson5),
        '/configuration-package/package.json': JSON.stringify(packageJson6),
      }

      vol.fromJSON(dir)

      await clearPackageJson('/configuration-package', '/target')

      expect(
        JSON.parse(vol.readFileSync('/target/package.json', 'utf8')),
      ).toMatchSnapshot()
    })
  })
})
