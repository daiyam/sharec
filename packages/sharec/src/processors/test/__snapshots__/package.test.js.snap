// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processors > package > processPackageJson should merge exist package.json with upcoming from configs 1`] = `
Object {
  "config": Object {
    "commitizen": Object {
      "path": "cz-conventional-changelog",
    },
  },
  "dependencies": Object {
    "react": "16.0.0",
  },
  "devDependencies": Object {
    "@commitlint/cli": "^7.6.1",
    "@commitlint/config-conventional": "^7.6.0",
    "babel-eslint": "^10.0.1",
    "commitizen": "^3.1.1",
    "cz-conventional-changelog": "^2.1.0",
    "eslint": "^5.16.0",
    "eslint-config-prettier": "^4.2.0",
    "eslint-config-prettier-standard": "^2.0.0",
    "eslint-config-standard": "^12.0.0",
    "eslint-plugin-import": "^2.17.2",
    "eslint-plugin-node": "^9.0.1",
    "eslint-plugin-prettier": "^3.1.0",
    "eslint-plugin-promise": "^4.1.1",
    "eslint-plugin-standard": "^4.0.0",
    "husky": "^2.2.0",
    "lint-staged": "^8.1.6",
    "prettier": "^1.17.1",
  },
  "eslintConfig": Object {
    "env": Object {
      "jest": true,
      "node": true,
    },
    "extends": Array [
      "standard",
      "prettier",
      "prettier/standard",
    ],
    "parser": "babel-eslint",
    "plugins": Array [
      "prettier",
    ],
    "rules": Object {
      "prettier/prettier": Array [
        "error",
        Object {
          "semi": false,
          "singleQuote": true,
          "trailingComma": "all",
        },
      ],
    },
  },
  "eslintIgnore": Array [
    "/node_modules",
  ],
  "husky": Object {
    "hooks": Object {
      "commit-msg": "commitlint -E HUSKY_GIT_PARAMS",
      "pre-commit": "lint-staged",
    },
  },
  "lint-staged": Object {
    "src/**/*.{js,jsx,json}": Array [
      "eslint --fix",
      "prettier --write",
      "git add",
    ],
  },
  "name": "my-awesome-package",
  "prettier": Object {
    "semi": false,
    "singleQuote": true,
    "trailingComma": "all",
  },
  "scripts": Object {
    "commit": "",
    "lint": "eslint 'src/**/*.{js,jsx,json}'",
    "lint-staged": "lint-staged",
  },
  "sharec": Object {
    "injected": true,
  },
}
`;
